decentraland/explorer
# fix: Implement the needed loading feedback for BiW

This PR fix the issue: https://github.com/decentraland/unity-renderer/issues/177

### Improvements included
- A new **loading panel** (`BuilderInWorldLoadingView / BuilderInWorldLoadingController`) that will be showed when the Builder In World be activated and hidden when all the editor has been loaded. This panel will show a carousel of phrases (random tips) during the loading process, and will let us to cancel and exit from the loading panel in any moment.

![image](https://user-images.githubusercontent.com/64659061/112144574-b2459200-8bd9-11eb-9eaf-de333fdc6e7e.png)

- A **loading spinner** for each catalog item to indicate when its thumbnail is being loaded.

![image](https://user-images.githubusercontent.com/64659061/112145666-f08f8100-8bda-11eb-8e64-75d680f92ff3.png)

### Testing
1. Go to: https://explorer.decentraland.zone/branch/fix/Implement-the-needed-loading-feedback-for-BiW/index.html?ENV=org&ENABLE_BUILDER_IN_WORLD&position=100,100&realm=fenrir-amber
2. Activate the Builder In World editor pressing the "**K**" key.
3. See the **loading panel** that will be showed.
4. Once the editor has been loaded, click on **Catalog** button and select any category.
5. See the loading spinners that will be showed on each item that is being loaded.
